apps/src/templates/instructions/ReviewTab.jsx

```diff
@@ -249,47 +252,39 @@ class ReviewTab extends Component {
     );
   }
 
-  setReadyForReview(isReadyForReview) {
+  setReadyForReview(shouldBeReadyForReview) {
     this.setState({
       loadingReviewableState: true,
       errorSavingReviewableProject: false
     });
 
-    if (isReadyForReview) {
+    const errorResponse = () =>
+      this.setState({
+        errorSavingReviewableProject: true,
+        loadingReviewableState: false
+      });
+
+    const newState = {
+      isReadyForReview: shouldBeReadyForReview,
+      errorSavingReviewableProject: false,
+      loadingReviewableState: false
+    };
+
+    if (shouldBeReadyForReview) {
       this.dataApi
         .enablePeerReview()
-        .done(data => {
+        .done(data =>
           this.setState({
-            reviewableProjectId: data.id,
-            isReadyForReview: true,
-            errorSavingReviewableProject: false,
-            loadingReviewableState: false
-          });
-        })
-        .fail(() => {
-          this.setState({
-            isReadyForReview: false,
-            errorSavingReviewableProject: true,
-            loadingReviewableState: false
-          });
-        });
+            ...newState,
+            reviewableProjectId: data.id
+          })
+        )
+        .fail(errorResponse);
     } else {
       this.dataApi
         .disablePeerReview(this.state.reviewableProjectId)
-        .done(() => {
-          this.setState({
-            isReadyForReview: false,
-            errorSavingReviewableProject: false,
-            loadingReviewableState: false
-          });
-        })
-        .fail(() => {
-          this.setState({
-            isReadyForReview: true,
-            errorSavingReviewableProject: true,
-            loadingReviewableState: false
-          });
-        });
+        .done(() => this.setState(newState))
+        .fail(errorResponse);
```

Comment:
Looks like we used to set `isReadyForReview` to true on fail here, was that unnecessary/wrong?

Reply:
I removed it because from what I could see, setting isReadyForReview here was a no-op. This function is invoked by the previous function, `renderReadyForReviewCheckbox`. It would always call this function with `setReadyForReview(!this.state.isReadyForReview)`. Tracing the logic through this function, the fail state would always call the equivalent of `this.setState({isReadyForReview: this.state.isReadyForReview})`. That being said, I could use a second pair of eyes here as I may have read it wrong.

Reply:
Took a look and that sounds right to me too, nice cleanup!

Reply:
Yep, just to provide some additional context, I believe the intention here was to retain the previous state of `isReadyForReview` if the API call fails, but since we're not explicitly changing the state on fail here, it should work the same way (and is much more streamlined 😄 )